frontend/src/store/modules/product.js

```diff
@@ -1,27 +1,53 @@
-const product = {
+import axios from 'axios';
 
+const product = {
     namespaced: true,
-
     state: { 
-        newProduct: undefined
+        newProduct: undefined,
+        allProducts: [],
     },
 
     mutations: {
         setNewProduct (state, product) {
             state.newProduct = product
-        }
+            state.allProducts.push(product)
+        },
+        updateProducts(state, products) {
+            state.allProducts = products
+        },
+        removeProductByIndex(state, idx) {
+          delete state.allProducts[idx]
```

Comment:
Small nitpick, but the `delete` keyword in JS has extremely bad performance. You're totally OK to use this here, but just wanted to call this out. This only really becomes a problem when you're performing this operation many times (i.e. cycling through and deleting a whole bunch of products at one time in rapid succession)